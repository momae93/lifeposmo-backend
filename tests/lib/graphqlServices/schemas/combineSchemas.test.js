const combineSchemas = require('../../../../src/lib/graphqlServices/schemas/combineSchemas');

describe('[LIB][GRAPHQL][RESOLVERS] : combineSchemas', () => {
  it('should match snapshot lib graphql combineSchemas', async () => {
    const schemas = combineSchemas();

    expect(schemas).toMatchSnapshot();
  });
});
