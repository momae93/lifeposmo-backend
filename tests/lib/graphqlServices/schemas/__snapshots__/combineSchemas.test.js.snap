// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[LIB][GRAPHQL][RESOLVERS] : combineSchemas should match snapshot lib graphql combineSchemas 1`] = `
Array [
  Object {
    "definitions": Array [
      Object {
        "description": undefined,
        "directives": Array [],
        "fields": Array [
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "_",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Boolean",
              },
            },
          },
        ],
        "interfaces": Array [],
        "kind": "ObjectTypeDefinition",
        "name": Object {
          "kind": "Name",
          "value": "Query",
        },
      },
      Object {
        "description": undefined,
        "directives": Array [],
        "fields": Array [
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "_",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Boolean",
              },
            },
          },
        ],
        "interfaces": Array [],
        "kind": "ObjectTypeDefinition",
        "name": Object {
          "kind": "Name",
          "value": "Mutation",
        },
      },
    ],
    "kind": "Document",
    "loc": Object {
      "end": 73,
      "start": 0,
    },
  },
  Object {
    "definitions": Array [
      Object {
        "description": undefined,
        "directives": Array [],
        "fields": Array [
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "id",
            },
            "type": Object {
              "kind": "NonNullType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "ID",
                },
              },
            },
          },
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "username",
            },
            "type": Object {
              "kind": "NonNullType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "String",
                },
              },
            },
          },
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "password",
            },
            "type": Object {
              "kind": "NonNullType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "String",
                },
              },
            },
          },
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "firstname",
            },
            "type": Object {
              "kind": "NonNullType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "String",
                },
              },
            },
          },
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "lastname",
            },
            "type": Object {
              "kind": "NonNullType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "String",
                },
              },
            },
          },
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "description",
            },
            "type": Object {
              "kind": "NonNullType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "String",
                },
              },
            },
          },
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "birthdate",
            },
            "type": Object {
              "kind": "NonNullType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "Date",
                },
              },
            },
          },
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "isMale",
            },
            "type": Object {
              "kind": "NonNullType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "Boolean",
                },
              },
            },
          },
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "postsWritten",
            },
            "type": Object {
              "kind": "ListType",
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "Post",
                  },
                },
              },
            },
          },
        ],
        "interfaces": Array [],
        "kind": "ObjectTypeDefinition",
        "name": Object {
          "kind": "Name",
          "value": "User",
        },
      },
      Object {
        "directives": Array [],
        "fields": Array [
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "users",
            },
            "type": Object {
              "kind": "ListType",
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "User",
                  },
                },
              },
            },
          },
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "id",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "Int",
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "user",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "User",
              },
            },
          },
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "me",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "User",
              },
            },
          },
        ],
        "interfaces": Array [],
        "kind": "ObjectTypeExtension",
        "name": Object {
          "kind": "Name",
          "value": "Query",
        },
      },
      Object {
        "directives": Array [],
        "fields": Array [
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "username",
                },
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "password",
                },
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "firstname",
                },
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "lastname",
                },
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "description",
                },
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "isMale",
                },
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "Boolean",
                    },
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "birthdate",
                },
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "Date",
                    },
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "createUser",
            },
            "type": Object {
              "kind": "NonNullType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "User",
                },
              },
            },
          },
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "id",
                },
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "ID",
                    },
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "deleteUser",
            },
            "type": Object {
              "kind": "NonNullType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "Boolean",
                },
              },
            },
          },
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "idUser",
                },
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "ID",
                    },
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "idFavoriteUser",
                },
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "ID",
                    },
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "followUser",
            },
            "type": Object {
              "kind": "NonNullType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "User",
                },
              },
            },
          },
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "idFavorite",
                },
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "ID",
                    },
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "unfollowUser",
            },
            "type": Object {
              "kind": "NonNullType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "Boolean",
                },
              },
            },
          },
        ],
        "interfaces": Array [],
        "kind": "ObjectTypeExtension",
        "name": Object {
          "kind": "Name",
          "value": "Mutation",
        },
      },
    ],
    "kind": "Document",
    "loc": Object {
      "end": 766,
      "start": 0,
    },
  },
  Object {
    "definitions": Array [
      Object {
        "description": undefined,
        "directives": Array [],
        "fields": Array [
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "id",
            },
            "type": Object {
              "kind": "NonNullType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "ID",
                },
              },
            },
          },
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "title",
            },
            "type": Object {
              "kind": "NonNullType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "String",
                },
              },
            },
          },
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "description",
            },
            "type": Object {
              "kind": "NonNullType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "String",
                },
              },
            },
          },
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "solution",
            },
            "type": Object {
              "kind": "NonNullType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "String",
                },
              },
            },
          },
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "totalClaps",
            },
            "type": Object {
              "kind": "NonNullType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "Int",
                },
              },
            },
          },
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "idAuthor",
            },
            "type": Object {
              "kind": "NonNullType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "Int",
                },
              },
            },
          },
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "author",
            },
            "type": Object {
              "kind": "NonNullType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "User",
                },
              },
            },
          },
        ],
        "interfaces": Array [],
        "kind": "ObjectTypeDefinition",
        "name": Object {
          "kind": "Name",
          "value": "Post",
        },
      },
      Object {
        "directives": Array [],
        "fields": Array [
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "posts",
            },
            "type": Object {
              "kind": "ListType",
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "Post",
                  },
                },
              },
            },
          },
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "id",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "Int",
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "post",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Post",
              },
            },
          },
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "idAuthor",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "Int",
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "postsWrittenBy",
            },
            "type": Object {
              "kind": "ListType",
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "Post",
                  },
                },
              },
            },
          },
        ],
        "interfaces": Array [],
        "kind": "ObjectTypeExtension",
        "name": Object {
          "kind": "Name",
          "value": "Query",
        },
      },
      Object {
        "directives": Array [],
        "fields": Array [
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "title",
                },
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "description",
                },
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "solution",
                },
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "idAuthor",
                },
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "Int",
                    },
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "createPost",
            },
            "type": Object {
              "kind": "NonNullType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "Post",
                },
              },
            },
          },
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "id",
                },
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "ID",
                    },
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "deletePost",
            },
            "type": Object {
              "kind": "NonNullType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "Boolean",
                },
              },
            },
          },
        ],
        "interfaces": Array [],
        "kind": "ObjectTypeExtension",
        "name": Object {
          "kind": "Name",
          "value": "Mutation",
        },
      },
    ],
    "kind": "Document",
    "loc": Object {
      "end": 475,
      "start": 0,
    },
  },
  Object {
    "definitions": Array [
      Object {
        "description": undefined,
        "directives": Array [],
        "kind": "ScalarTypeDefinition",
        "name": Object {
          "kind": "Name",
          "value": "Date",
        },
      },
    ],
    "kind": "Document",
    "loc": Object {
      "end": 17,
      "start": 0,
    },
  },
]
`;
